flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerNG.java

```diff
@@ -93,6 +94,8 @@ ExecutionState requestPartitionState(
 
     ExecutionGraphInfo requestJob();
 
+    CheckpointStatsSnapshot requestCheckpointStats();
```

Comment:
Why do we need this? Can JM access this through the ExecutionGraphInfo?

Reply:
Same as above, Yes JM can access this via ExecutionGraphInfo, but that means we would incur the cost of recursive copy of the ExecutionGraph when all we need is the CheckpointStatsSnapshot object.

Reply:
👍 can you please add a javadoc with an explanation?

Reply:
Yep can do.